File: services/website/shared/src/main/java/nl/runnable/gigmatch/model/Skill.java
package nl.runnable.gigmatch.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Skill {
    private UUID uuid;

    private String name;

    private String slug;
}

File: services/match-service/domain/src/main/kotlin/nl/runnable/gigmatch/domain/vacancy/Skill.kt
package nl.runnable.gigmatch.domain.vacancy

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class Skill @JsonCreator constructor(
    @JsonProperty("id")
    val id: SkillId,
    @JsonProperty("name")
    val name: String,
    @JsonProperty("slug")
    val slug: String,
) {
    init {
        require(name.isNotBlank()) {
            "Name cannot be blank"
        }

        require(slug.isNotBlank()) {
            "Slug cannot be blank"
        }
    }
}
